perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -2352,17 +2387,20 @@ public void updateFormItem(PerunSession sess, ApplicationFormItem item) throws P
 
 				}
 
-				// We do require value from IDP (federation) if attribute is supposed to be pre-filled and item is required and not editable to users
-				if ((itemW.getPrefilledValue() == null || itemW.getPrefilledValue().isEmpty()) && itemW.getFormItem().isRequired() &&
-						(Type.FROM_FEDERATION_HIDDEN.equals(itemW.getFormItem().getType()) || Type.FROM_FEDERATION_SHOW.equals(itemW.getFormItem().getType()))) {
+			}
+		}
 
-					if (URN_USER_DISPLAY_NAME.equals(item.getPerunDestinationAttribute())) {
-						log.error("Couldn't resolve displayName from: {}, parsedNames were: {}", federValues, parsedName);
-					}
+		Map<Integer, ApplicationFormItemWithPrefilledValue> allItemsByIds = itemsWithValues.stream()
+				.collect(toMap(item -> item.getFormItem().getId(), Function.identity()));
 
-					itemsWithMissingData.add(itemW);
+		for (ApplicationFormItemWithPrefilledValue itemW : itemsWithValues) {
+			// We do require value from IDP (federation) if attribute is supposed to be pre-filled and item is required and not editable to users
+			if ((itemW.getPrefilledValue() == null || itemW.getPrefilledValue().isEmpty()) &&
+					itemW.getFormItem().isRequired() && isItemHidden(itemW, allItemsByIds)) {
```

Comment:
What about disabled items?

Reply:
Good point, I added a check for that as well.